cloud/passportauth/src/auth/PassportSetup.ts

```diff
@@ -0,0 +1,79 @@
+import * as express from 'express';
+import * as session from 'express-session';
+import {SessionOptions} from 'express-session';
+import * as passport from 'passport';
+import {Strategy} from 'passport-local';
+import {UserSec} from '../user/UserSec';
+
+export interface Setup {
+  init(app: express.Express, sessionOpts: SessionOptions): void;
+  setStrategy(): void;
+  setSerializeUser(): void;
+  setDeserializeUser(): void;
+}
+
+export class PassportSetup implements Setup {
+  protected userSec: UserSec;
+
+  constructor(protected readonly userSecService: UserSec) {
+    this.userSec = userSecService;
+  }
+
+  /**
+   * Initializes passport authentication on an Express application.
+   * This initializes the Express session, passport as well as
+   * setting the local strategy, serializeUser and deserializeUser function.
+   *
+   * @param app {express.Express} An express app used by the application.
+   * @param sessionOpts {SessionOptions} Configuration used by Express session
+   */
+  public init(app: express.Express, sessionOpts: SessionOptions) {
+    app.use(session(sessionOpts));
+    app.use(passport.initialize());
+    app.use(passport.session());
+
+    this.setStrategy();
+    this.setSerializeUser();
+    this.setDeserializeUser();
+  }
+
+  /**
+   * Sets the local strategy used by passport authentication.
+   */
+  public setStrategy() {
```

Comment:
As agreed - let's allow people to setup this. 

Expose our strategy as constant so people can do something like:

`setStrategy(DEFAULT_STRATEGY)`